### PR TITLE
✨ cnquery status show latest version available

### PR DESCRIPTION
**Pull request by mariuskimmina**
This adds a new output to `cnquery status` which shows the version number of the latest release available on github and shows a hint that a newer version is available if it the currently installed version and the latest version don't match.

```
./cnquery status
→ no Mondoo configuration file provided, using defaults
→ Platform:             ubuntu
→ Version:              22.04
→ Hostname:             mondoopad
→ IP:                   192.168.178.32
→ Time:                 2023-11-01T13:36:01+01:00
→ Version:              unstable (API Version: unstable)
→ Latest Version:       9.4.0
! A newer version is available
→ API ConnectionConfig: https://us.api.mondoo.com
→ API Status:           SERVING
→ API Time:             2023-11-01T12:36:02Z
→ API Version:          9
```

Related to: https://github.com/mondoohq/cnspec/issues/887

### CODE REVIEW

File: cnquery.go [hunk 1]
```diff
@@ -42,6 +47,47 @@ func GetVersion() string {
...
+	}
+
+	return release.Name, nil
+}
+
+// GetLatestVersion returns the latest version available on Github
+func GetLatestVersion() string {
+	releaseName, err := GetLatestReleaseName()
+	if err != nil {
+		panic(err)
```
Comment (preslavgerchev) [inline cnquery.go]: i dont think we want to panic here, failing to get latest version should not be a cause panic on the status call, we should just push the err through

File: cnquery.go [hunk 2]
```diff
@@ -42,6 +47,47 @@ func GetVersion() string {
...
 }
 
+// Release represents a GitHub release
+type Release struct {
+	Name string `json:"name"`
+}
+
+// GetLatestReleaseName fetches the name of the latest release from the specified GitHub repository
+func GetLatestReleaseName() (string, error) {
+	url := "https://api.github.com/repos/mondoohq/cnquery/releases/latest"
```
Comment (preslavgerchev) [inline cnquery.go]: nit: can we make this a param and rename to `defaultUrl`. we can reuse this func for cnspec as well, if needed

File: apps/cnquery/cmd/status.go [hunk 3]
```diff
@@ -149,37 +149,42 @@ type ClientStatus struct {
...
 	if s.Upstream.API.Version != cnquery.APIVersion() {
 		log.Warn().Msg("API versions do not match, please update the client")
 	}
 
 	if len(s.Upstream.Features) > 0 {
-		log.Info().Msg("Features:\t" + strings.Join(s.Upstream.Features, ","))
+		log.Info().Msg("Features:\t\t" + strings.Join(s.Upstream.Features, ","))
 	}
 
 	if s.Client.ParentMrn == "" {
```
Comment (preslavgerchev) [inline apps/cnquery/cmd/status.go]: shouldn't this check be `!=`?

File: cnquery.go (hunk 1 above)
Comment (mariuskimmina) [reply cnquery.go]: Right, changed it

File: apps/cnquery/cmd/status.go (hunk 3 above)
Comment (mariuskimmina) [reply apps/cnquery/cmd/status.go]: Hm, as long as the numbers can't be negative it behaves the same I think

Comment (preslavgerchev) [reply apps/cnquery/cmd/status.go]: I meant the `s.Client.ParentMrn == ""` block, it should probably be `s.Client.ParentMrn != ""`,

there is no point in showing an empty parent mrn

File: apps/cnquery/cmd/status.go [hunk 4]
```diff
@@ -149,37 +149,50 @@ type ClientStatus struct {
 func (s Status) RenderCliStatus() {
 	if s.Client.Platform != nil {
 		agent := s.Client
-		log.Info().Msg("Platform:\t" + agent.Platform.Name)
-		log.Info().Msg("Version:\t" + agent.Platform.Version)
-		log.Info().Msg("Hostname:\t" + agent.Hostname)
-		log.Info().Msg("IP:\t\t" + agent.IP)
+		log.Info().Msg("Platform:\t\t" + agent.Platform.Name)
```
Comment (preslavgerchev) [inline apps/cnquery/cmd/status.go:152]: why is this changed to a double tab?

Comment (mariuskimmina) [reply apps/cnquery/cmd/status.go:152]: They looked more properly lined up this way for me.

File: apps/cnquery/cmd/status.go (hunk 3 above)
Comment (mariuskimmina) [reply apps/cnquery/cmd/status.go]: Oh, you are right, changed it

**Review by preslavgerchev — APPROVED** [review]
